### DISCUSSION TITLE
How are stopped/discarded actors handled in the visualiser?

### DISCUSSION THREAD

**Discussion by JamieMason** [statelyai/xstate#2647] [Q&A] [answered]
I have a question around the expected visualiser behaviour when stopping actors if anyone can help please. I've made an example and tried to explain my question at https://d8x95.sse.codesandbox.io/. 

The question is: should the visualiser dropdown only show child machines/actors while they're alive? or does it leave them around to help with debugging?

You can find the source at https://codesandbox.io/s/runtime-forest-d8x95?file=/parent/machine.js. 

Thanks.

**Comment by farskid** [comment]
@JamieMason the sandbox refers to the old inspector. Did you mean to use the new visualizer at https://stately.ai/viz?inspect ?

**Reply by JamieMason** [reply]
I wanted to avoid complicating things as I've been having some trouble with the new visualiser. I'll try to create a separate repro for that, but in short what happens is the initial visualisation is fine, but choosing another spawned actor onwards results in blank visualisations.

**Reply by farskid** [reply]
https://github.com/statelyai/xstate/discussions/2647#discussioncomment-1337123

**Reply by JamieMason** [reply]
Thanks, I think this answers my question then on what I should expect. I wasn't sure if I was actually stopping them correctly or not 👍🏻

**Comment by farskid** [comment, marked as answer]
I'm not sure about the old inspector but with the new visualizer/inspector (soon after the next release statelyai/xstate-viz#214), stopped actors will be shown as struck through while kept in the list for debugging purposes.
![image](https://user-images.githubusercontent.com/8332043/133573284-9fe27480-06ce-4ff6-b92b-352e151f0a12.png)
